initrd/etc/functions

```diff
@@ -440,6 +441,177 @@ update_checksums() {
+				TEXT+="$UPDATE_TEXT"
+			fi
+		fi
+
+		# If update of checksum is not optionally selected, show stern warning
+		if [ "$UPDATE" == "n" ]; then
+			DEBUG "Checksums update not selected by caller, showing stern warning"
+			whiptail $BG_COLOR_ERROR --title 'ERROR: Boot Hash Mismatch' \
+				--msgbox "$TEXT" 0 80
+			BG_COLOR_MAIN_MENU=$BG_COLOR_ERROR
+			return 1
+		else
+			DEBUG "Checksums update selected by caller, showing stern warning with option to update"
+			if (whiptail $BG_COLOR_ERROR --title 'ERROR: Boot Hash Mismatch' --yesno "$TEXT" 0 80); then
+				if update_checksums; then
+					BG_COLOR_MAIN_MENU=""
+					return 0
+				else
+					whiptail $BG_COLOR_ERROR --title 'ERROR' \
+						--msgbox "Failed to update checksums / sign default config" 0 80
+					BG_COLOR_MAIN_MENU=$BG_COLOR_ERROR
+                			return 1
+				fi
+			fi
+		fi
+	fi
+}
+
+report_integrity_measurements() {
+	#check for GPG key in keyring
+	GPG_KEY_COUNT=$(gpg -k 2>/dev/null | wc -l)
+	if [ $GPG_KEY_COUNT -ne 0 ]; then
+		# Check and report TOTP
+		# update the TOTP code every thirty seconds
+		date=$(date "+%Y-%m-%d %H:%M:%S %Z")
+		seconds=$(date "+%s")
+		half=$(expr \( $seconds % 60 \) / 30)
+		if [ "$CONFIG_TPM" != "y" ]; then
+			TOTP="NO TPM"
+		elif [ "$half" != "$last_half" ]; then
+			last_half=$half
+			TOTP=$(unseal-totp) >/dev/null 2>&1 || TOTP="TPM UNSEALING OF TOTP/HOTP SECRET FAILED"
+		fi
+
+		# Check and report on HOTP status
+		if [ -x /bin/hotp_verification ]; then
+			HOTP=$(unseal-hotp) >/dev/null 2>&1
+			enable_usb
+			while ! hotp_verification info >/dev/null 2>&1; do
+				whiptail $CONFIG_WARNING_BG_COLOR --title 'WARNING: Please insert your HOTP enabled USB Security dongle' --msgbox "Your HOTP enabled USB Security dongle was not detected.\n\nPlease remove it and insert it again." 0 80
+			done
+			# Don't output HOTP codes to screen, so as to make replay attacks harder
+			hotp_verification check $HOTP
+			case "$?" in
+			0)
+				HOTP="Success"
+				;;
+			4)
+				HOTP="Invalid code"
+				MAIN_MENU_BG_COLOR=$CONFIG_ERROR_BG_COLOR
+				;;
+			*)
+				HOTP="HOTP VERIFICATION FAILED. Returned error: $?"
+				MAIN_MENU_BG_COLOR=$CONFIG_WARNING_BG_COLOR
+				;;
+			esac
+		else
+			HOTP='N/A'
+		fi
+		# Check for detached signed digest and report on /boot integrity status
+		SIGNED="OK"
+		(check_config /boot) || SIGNED="ALTERED"
+
+		TMP_HASH_FILE="/tmp/kexec/kexec_hashes.txt"
+
+		if (cd /boot && sha256sum -c "$TMP_HASH_FILE" >/tmp/hash_output); then
+			HASH="OK"
+		else
+			HASH="ALTERED"
+		fi
+
+		#Show results
+
+		#If the system is in a bad state, show the user a warning
+		whiptail $MAIN_MENU_BG_COLOR --title "Measured Integrity Report" --msgbox "$date\n\nTOTP: $TOTP\nHOTP: $HOTP\nBOOT INTEGRITY SIGNATURE: $SIGNED\nBOOT INTEGRITY HASHES: $HASH\n\nPress OK to continue with detailed information in case of errors" 0 80
+
```

Comment:
IMO:

* All the CAPS introduced in the error text, as well as  the much longer labels for integrity, make this difficult to read.  There's no longer any separation between the "labels" and "content" for errors (screenshot below)
* I think the separation between "signature" and "hashes" should be an implementation detail of Heads - that's how we authenticate /boot.  The normal user flow should just say whether /boot integrity was authenticated or not, so it's easier to understand if you are not an expert in Heads.  (I'd be OK with providing detailed information as an optional thing or on the console, etc. to cater to advanced users too, but putting it front-and-center IMO makes it a requirement to be an knowledgeable in Heads internals to use Heads.)

Re: the caps, this is what I was greeted with when firing this up in qemu:

![Screenshot_20231012_171341](https://github.com/osresearch/heads/assets/109107914/c72eed32-16c6-48b7-9e2e-bb8917245125)

From a glance it is really hard to parse this and know what to look at.  The text-only prompt is a bit limiting but I'd suggest:
* Don't use all caps for the errors, this doesn't add anything and just makes it harder to read
* Go back to just "INTEGRITY: <status>" instead of the last two lines, which would make the labels all similar-length again and reduce a lot of verbosity
* Take out the colon from the HOTP error so the colons form a bit of separation between the labels and contents

Reply:
Oh and also, it did not show me any detailed information when I pressed Enter :-/